### ISSUE TITLE
nuxt3.8.X + tauri : "Cannot read properties of undefined (reading 'entries')"

### ISSUE THREAD

**Issue by flapili** [nuxt/nuxt#24625]
### Environment

nuxi info :
```
- Operating System: `Darwin`
- Node Version:     `v18.18.2`
- Nuxt Version:     `3.8.2`
- CLI Version:      `3.10.0`
- Nitro Version:    `2.8.1`
- Package Manager:  `pnpm@8.10.5`
- Builder:          `-`
- User Config:      `devtools`, `ssr`, `devServer`
- Runtime Modules:  `-`
- Build Modules:    `-`
```

tauri info:
```
✔] Environment
    - OS: Mac OS 14.1.2 X64
    ✔ Xcode Command Line Tools: installed
    ✔ rustc: 1.74.0 (79e9716c9 2023-11-13)
    ✔ cargo: 1.74.0 (ecb9851af 2023-10-18)
    ✔ rustup: 1.26.0 (5af9b9484 2023-04-05)
    ✔ Rust toolchain: stable-x86_64-apple-darwin (default)
    - node: 18.18.2
    - pnpm: 8.10.5
    - yarn: 1.22.19
    - npm: 9.8.1
    - bun: 1.0.0

[-] Packages
    - tauri [RUST]: 2.0.0-alpha.18
    - tauri-build [RUST]: 2.0.0-alpha.12
    - wry [RUST]: 0.34.2
    - tao [RUST]: 0.23.0
    - tauri-cli [RUST]: 2.0.0-alpha.9
    - @tauri-apps/api : not installed!
    - @tauri-apps/cli [NPM]: 2.0.0-alpha.18

[-] App
    - build-type: bundle
    - CSP: unset
    - distDir: 
    - devPath: http://localhost:1420/
    - framework: Vue.js (Nuxt)
    - bundler: Webpack

[-] iOS
    - Developer Teams: retracted
```

### Reproduction

https://github.com/flapili/repro-nuxt3.8-tauri

### Describe the bug

<img width="487" alt="image" src="https://github.com/nuxt/nuxt/assets/29481952/1cb79b7a-8af5-4cbb-9948-baab5abb3a45">

### Additional context

I thinks it's due to how tauri handle http ( proxied via `tauri.localhost/`) ? also if I downgrade nuxt to 3.7.4 it work

### Logs

```shell-script
nuxt.js?v=d306a4ef:97 [nuxt] error caught during app initialization TypeError: Cannot read properties of undefined (reading 'entries')
    at _sortRoutesMap (index.mjs?v=d306a4ef:218:16)
    at _matchRoutes (index.mjs?v=d306a4ef:200:30)
    at Object.matchAll (index.mjs?v=d306a4ef:158:25)
    at getRouteRules (manifest.js?v=d306a4ef:27:30)
    at async manifest-route-rule.js?v=d306a4ef:8:89
    at async router.js?v=d306a4ef:126:26
```

**Comment by manniL** [comment]
This is probably related to the manifest.json and a relative baseUrl.

**Comment by flapili** [comment]
if you need any information or extra test don't hesitate to @ me here or on the discord (same username as github), like using a particular commit ?

**Comment by 19119294973** [comment]
@flapili Can you tell me the specific solution? How to close appManifest?

**Comment by flapili** [comment]
> @flapili Can you tell me the specific solution? How to close appManifest?
Hi,
```ts
  experimental: {
    appManifest: false,
  },
```
in nuxt config

**Comment by ashokgelal** [comment]
I ran into this issue today. When setting `appManifest` to `false` it works but then now the routeRules defined in nuxt config is getting ignored. `appManifest` seems to have introduced all these issues. If it was an experimental feature, why not make in an opt-in rather than opt-out?

**Comment by ashokgelal** [comment]
Sorry but I cannot repro at all but some of our users can. And interestingly only on Windows not on Mac or Linux. I gave them a new build with appManifest set to false and they are saying that it works [1]. It worked for majority of the users though. I've asked for the logs and it seems to crash almost immediately so nothing in logs. I wish I could repro this locally. 

"we stabilised it". It being under experimental is what made me think that it is still experimental. 

[1] 
![image](https://github.com/nuxt/nuxt/assets/401055/8d8ce3e0-3e62-4bb5-b0e6-4e25f8f980a3)
